model/settings/service.go

```diff
@@ -2,53 +2,125 @@ package settings
+}
+
 // SettingsService handle the business logic around "settings".
 //
 // This service handle 2 structured documents present in [consts.Settings]
 // - The "instance settings" ([consts.InstanceSettingsID])
 // - The "bitwarden settings" ([consts.BitwardenSettingsID]) (#TODO)
 type SettingsService struct {
+	emailer  emailer.Emailer
+	instance instance.Service
+	token    token.Service
+	storage  Storage
 }
 
 // NewService instantiates a new [SettingsService].
-func NewService() *SettingsService {
-	return &SettingsService{}
+func NewService(
+	emailer emailer.Emailer,
+	instance instance.Service,
+	token token.Service,
+	storage Storage,
+) *SettingsService {
+	return &SettingsService{emailer, instance, token, storage}
+}
+
+// PublicName returns the settings' public name or a default one if missing
+func (s *SettingsService) PublicName(db prefixer.Prefixer) (string, error) {
+	doc, err := s.storage.getInstanceSettings(db)
+	if err != nil {
+		return "", err
+	}
+	publicName, _ := doc.M["public_name"].(string)
+	// if the public name is not defined, use the instance's domain
+	if publicName == "" {
+		split := strings.Split(db.DomainName(), ".")
+		publicName = split[0]
+	}
+	return publicName, nil
 }
 
+// GetInstanceSettings allows for fetch directly the [consts.InstanceSettingsID] couchdb document.
 func (s *SettingsService) GetInstanceSettings(inst prefixer.Prefixer) (*couchdb.JSONDoc, error) {
-	doc := &couchdb.JSONDoc{}
+	return s.storage.getInstanceSettings(inst)
+}
+
+// SetInstanceSettings allows a set directly the [consts.InstanceSettingsID] couchdb document.
+func (s *SettingsService) SetInstanceSettings(inst prefixer.Prefixer, doc *couchdb.JSONDoc) error {
+	return s.storage.setInstanceSettings(inst, doc)
+}
 
-	err := couchdb.GetDoc(inst, consts.Settings, consts.InstanceSettingsID, doc)
+type UpdateEmailCmd struct {
+	Passphrase []byte
+	Email      string
+}
+
+// StartEmailUpdate will start the email updating process.
+//
+// This process consists of:
+// - Validating the user with a password
+// - Sending a validation email to the new address with a validation link
+// - Confirm the new address when the validation link is called
+func (s *SettingsService) StartEmailUpdate(inst *instance.Instance, cmd *UpdateEmailCmd) error {
+	err := s.instance.CheckPassphrase(inst, cmd.Passphrase)
 	if err != nil {
-		return nil, err
+		return fmt.Errorf("failed to check passphrase: %w", err)
 	}
 
-	doc.Type = consts.Settings
+	settings, err := s.storage.getInstanceSettings(inst)
+	if err != nil {
+		return fmt.Errorf("failed to fetch the instance: %w", err)
+	}
 
-	return doc, nil
-}
+	publicName, err := s.PublicName(inst)
+	if err != nil {
+		return fmt.Errorf("failed to retrieve the instance settings: %w", err)
+	}
 
-func (s *SettingsService) SetInstanceSettings(inst prefixer.Prefixer, doc *couchdb.JSONDoc) error {
-	// TODO: Validate input
+	settings.M["pending_email"] = cmd.Email
```

Comment:
I don't see where this change is persisted to CouchDB

Reply:
I just added now.